packages/core/src/tools/error.ts

```diff
@@ -32,26 +52,48 @@ export const enum ErrorHandling {
 
 export type ErrorSource = typeof ErrorSource[keyof typeof ErrorSource]
 
-export function formatUnknownError(
-  stackTrace: StackTrace | undefined,
-  errorObject: any,
-  nonErrorPrefix: string,
+type RawErrorParams = {
+  stackTrace?: StackTrace
+  error: unknown
   handlingStack?: string
-) {
-  if (!stackTrace || (stackTrace.message === undefined && !(errorObject instanceof Error))) {
+  startClocks: ClocksState
+  nonErrorPrefix: string
+  source: ErrorSource
+  handling?: ErrorHandling
```

Comment:
<!-- Suggestions are specific requests to improve the subject under review. It is assumed that we all want to do what's best, so these comments are never dismissed as “mere suggestions”, but are taken seriously.  -->
**💬 suggestion:** ‏from usages
```suggestion
  handling: ErrorHandling
```